conans/client/loader.py

```diff
@@ -199,6 +200,10 @@ def _initialize_conanfile(conanfile, profile):
         conanfile.initialize(tmp_settings, profile.buildenv)
         conanfile.conf = profile.conf.get_conanfile_conf(ref_str)
 
+        # Initialize the generic remotes
+        if self._remotes_registry:
+            conanfile.generic_remotes = [r for r in self._remotes_registry.list() if r.generic]
```

Comment:
I'd rather pass here the ``generic_remotes`` already processed in the origin, and not listing and filtering at each conanfile loading.